road/scatter.py

```diff
@@ -1,109 +1,39 @@
-y_field2 = config['PLOT2']['YField']
-name2 = config['PLOT2']['Name']
-
-dashboard_num = config['YAML']['Dashboard_number']
-
-if len(dbf_files) != len(output_columns):
-    raise ValueError("The number of DBF files and output column names do not match.")
-
-est_df = filter_and_aggregate(excel_file_path, sheet_name, dbf_directory, dbf_files, column_names, output_columns, extra_columns)
-
-obs_df = pd.read_excel(excel_file_path, skiprows=1, sheet_name=sheet_name, usecols=obs_usecols)
-
-common_columns = obs_df.columns
-
-error = {}
-for col in common_columns:
-    error[col] = est_df[col] - obs_df[col]
-error_df = pd.DataFrame(error)
-error_squared_df = error_df.pow(2)
-error_percent_df = error_df.divide(obs_df).fillna(0)
-error_percent_df = error_percent_df.apply(lambda x: x * 100, axis=1)
-
-combined_df = pd.concat([est_df, obs_df, error_df, error_squared_df, error_percent_df], axis=1)
-Select_time_period_df = combined_df.filter(like=chosen_timeperoid)
-specific_columns_combined_df = combined_df[combined_df_cols]
-Select_time_period_loc_df = pd.concat([specific_columns_combined_df, Select_time_period_df], axis=1)
-calculation_cols = ['Estimated Volume', 'Observed Volume', 'Errors','Squared Errors', 'Percent Errors']
-Select_time_period_loc_df.columns = combined_df_cols + calculation_cols
-Select_time_period_loc_df = Select_time_period_loc_df.drop_duplicates(subset=['A', 'B'], keep='first')
-
-file_name = f"{chosen_timeperoid}_all_data.csv"
-csv_base_name = os.path.splitext(file_name)[0]
-Select_time_period_loc_df.to_csv(file_name, index=False)
-classification_col_types = Select_time_period_loc_df[classification_col].unique()
+def compute_and_save_errors(est_df, obs_df, chosen_timeperiod, combined_df_cols, classification_col):
+    # Compute errors
+    times = ['Daily','AM', 'MD', 'PM', 'EV', 'EA']
+    error = {col: est_df[col] - obs_df[col] for col in times}
+    error_df = pd.DataFrame(error)
+    error_squared_df = error_df.pow(2)
+    error_percent_df = error_df.divide(obs_df[times]).fillna(0)
+    error_percent_df = error_percent_df.apply(lambda x: x * 100, axis=1)
+
+    # Combine dataframes
+    combined_df = pd.concat([est_df, obs_df[times], error_df, error_squared_df, error_percent_df], axis=1)
+    Select_time_period_df = combined_df.filter(like=chosen_timeperiod)
+    specific_columns_combined_df = combined_df[combined_df_cols]
+    Select_time_period_loc_df = pd.concat([specific_columns_combined_df, Select_time_period_df], axis=1)
+
+    # Set column names
+    calculation_cols = ['Estimated Volume', 'Observed Volume', 'Errors', 'Squared Errors', 'Percent Errors']
+    Select_time_period_loc_df.columns = combined_df_cols + calculation_cols
+
+    # Drop duplicates
+    Select_time_period_loc_df = Select_time_period_loc_df.drop_duplicates(subset=['A', 'B'], keep='first')
+
+    # Save to CSV
+    file_name = f"{chosen_timeperiod}_Scatter_data.csv"
+    Select_time_period_loc_df.to_csv(file_name, index=False)

```

Comment:
perhaps move the save filepath to the config file? Also, do we not need to specify the directory here?

Reply:
I'm not too sure about the directory structure about this code so maybe flag this for later